packages/tas_keybind/init.lua

```diff
@@ -0,0 +1,79 @@
+tas = tas or {}
+tas._oldControls = {}
+
+---@alias tas.KeyListenerCallback fun(player: unknown, pressed: boolean):void
+---@alias Set<T> table<T, boolean>
+
+---@type table<tas.CONTROLS, Set<tas.KeyListenerCallback>>
+tas.keyListeners = {}
+---@enum tas.CONTROLS
+tas.CONTROLS = {
+	up = 0,
+	down = 1,
+	left = 2,
+	right = 3,
+	jump = 4,
+	aux1 = 5,
+	sneak = 6,
+	dig = 7,
+	place = 8,
+	zoom = 9,
+}
+
+local function saveControlsForPlayer(player)
```

Comment:
I don't think the name of this function is as accurate as it could be. We're returning the old and new controls so it should definitely start with `get`.

Also, I'm not sure we actually need to know what the old controls were, just the changes. Given this, I think we should do the following:
- Replace this signature with `getControlsChanges(playerName, newControls)`
- Keep the logic inside the function to get the old controls and overwrite them, but don't return them
- Return the xor of the new controls passed in and the old controls

This means that this function doesn't need to know about how we get the player name or the controls, and the caller already has the new controls as it needed to pass them in, so no need to return anything but the changes. I think this provides a better separation of concerns.

Also, the caller doesn't need to know that we're saving the controls anywhere, that mechanism is abstracted.